### ISSUE TITLE
Support additional/globbed per-module content

### ISSUE THREAD

**Issue by richardszalay** [richardszalay/helix-publishing-pipeline#18]
For content not marked as `Content` in the project, for whatever reason. Two primary use cases are:

1. Per-module generated assets (js/css)
2. Per-module unicorn files

**Comment by richardszalay** [comment]
"2" is going to be tricky as the individual modules need to be included in a module specific folder (eg. "Feature.Accounts") which require some huge assumptions made in HPP. 

Focus will be "1" for now, until a reasonable approach can be found for "2"

**Comment by richardszalay** [comment]
The preference is to be able to define patterns at the global level rather than as transforms of `@(HelixModulePaths)` (which would require far more knowledge of MSBuild than most developers will bother with).

Example A: We should able to include `./assets/js/**/*` from each module
Example B: We should be able to include `../serialization/**/*` from each module and publish it as `App_Data\Serialization\%(ModuleName)\**\*` (this is the tricky one)

In order to achieve this, again without advanced MSBuild knowledge, we need to start by deciding how we will define the patterns verbatim (ie. unexpanded). To do this, they need to either be properties or item metata. 

Properties don't really work since there's no way to associate the _target_ path of each pattern (at least, without it getting _really_ ugly).

This leaves us with item metadata, but we can't use any existing variable syntax (`$(Prop)`, `@(Item)` or `%(Metadata)`) without it's value being resolved when the item is defined.

One proposal could be to use an alternative syntax and then have a task clean it up and create items from it:

```xml
<ItemGroup>
  <AdditionalFilesForPackagingFromHelixModules Include="Unicorn">
    <Source>..\serialization\**\*</Source>
    <Target>App_Data\^(Layer)\^(Module)\^(RecursiveDir)\^(Filename)^(Extension)</Target>
  </AdditionalFilesForPackagingFromHelixModules>
```

There's a few things to note here:

1. We'd need to be able to specify metadata for each module, though we could include "Module" for free when not specified (as the Filename of the source module)
2. We'd need to merge in the module item metadata into each content item, but this should be easily achievable through batching

**Comment by richardszalay** [comment]
@coreyasmith @jeneaux  Which syntax would offend you least for defining per-module unicorn content?

Option A - alternate character (feel free to offer another replacement character if you simply don't like `^`):
```
<ItemGroup>
  <AdditionalFilesForPackagingFromHelixModules Include="Unicorn">
    <Source>..\serialization\**\*</Source>
    <Target>App_Data\^(Layer)\^(Module)\^(RecursiveDir)\^(Filename)^(Extension)</Target>
  </AdditionalFilesForPackagingFromHelixModules>
</ItemGroup>
```

Option B - manually escaped
```
<ItemGroup>
  <AdditionalFilesForPackagingFromHelixModules Include="Unicorn">
    <Source>..\serialization\**\*</Source>
    <Target>App_Data\%25(Layer)\%25(Module)\%25(RecursiveDir)\%25(Filename)%25(Extension)</Target>
  </AdditionalFilesForPackagingFromHelixModules>
</ItemGroup>
```

Option B - helper escaped (this one might not even work as the parentheses might get escaped too)
```
<ItemGroup>
  <AdditionalFilesForPackagingFromHelixModules Include="Unicorn">
    <Source>..\serialization\**\*</Source>
    <Target>$([MSBuild]::Escape('App_Data\%(Layer)\%(Module)\%(RecursiveDir)\%(Filename)%(Extension)')</Target>
  </AdditionalFilesForPackagingFromHelixModules>
</ItemGroup>
```

**Comment by jeneaux** [comment]
I also think option A is good.

**Comment by richardszalay** [comment]
Released in v1.3.0
